### PR TITLE
Show HUD message for why player cannot engulf something (and a few other messages)

### LINKED ISSUE #3664: If the player can't engulf something they attempted to engulf it should be clarified why
there should be some kind of text popup or something if the player cells ejects something it couldn't engulf, this would really help players understand why they couldn't engulf something

### PR DESCRIPTION
**Pull request by hhyyrylainen**
**Brief Description of What This PR Does**

Shows up to 4 messages that fade over time to the player about various things.

Also included a few code maintenance things by updating some deps and dependabot updates should start working for this repo again.

**Related Issues**

closes #3664

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

**Comment by hhyyrylainen** [comment]
I already opened two follow up issues for a bit tricky parts that I didn't really try to mash into this initial implementation (also I ran out of time for today anyway):
- https://github.com/Revolutionary-Games/Thrive/issues/4146
- https://github.com/Revolutionary-Games/Thrive/issues/4147

File: locale/en.po [hunk 1]
```diff
@@ -2275,10 +2275,14 @@ msgstr "HP:"
 msgid "HSV"
 msgstr "HSV"
 
+#: ../src/microbe_stage/gui/HUDMessages.cs:37
+msgid "HUD_MESSAGE_MULTIPLE"
+msgstr "{0} (x {1})"
```
Comment (Kasterisk) [inline locale/en.po:2280]: A bit strange to have a space after the x, isn't this usually written as (x2), (x10), and so on? We already have `SPECIES_N_TIMES` too.

File: locale/en.po [hunk 2]
```diff
@@ -3903,11 +3907,35 @@ msgstr "Normal"
...
+msgid "NOTICE_ENGULF_DAMAGE_FROM_TOXIN"
+msgstr "Damaged by toxins in ingested matter"
+
+#: ../src/microbe_stage/Microbe.Interior.cs:1456
+msgid "NOTICE_ENGULF_MISSING_ENZYME"
+msgstr "Cannot engulf target due to missing: {0}"
+
+#: ../src/microbe_stage/Microbe.Contact.cs:1778
+msgid "NOTICE_ENGULF_SIZE_TOO_SMALL"
+msgstr "Cannot engulf target due to it being too big compared to current size"
```
Comment (Kasterisk) [inline locale/en.po:3929]:
```suggestion
msgstr "Cannot engulf target due to it being too big compared to current cell size"
```

File: src/microbe_stage/Microbe.Contact.cs [hunk 3]
```diff
@@ -1756,6 +1768,14 @@ private void CheckStartEngulfingOnCandidate(IEngulfable engulfable)
...
             OnEngulfmentStorageFull?.Invoke(this);
+
+            OnNoticeMessage?.Invoke(this,
+                new SimpleHUDMessage(TranslationServer.Translate("NOTICE_ENGULF_STORAGE_FULL")));
+        }
+        else if (EngulfSize < engulfable.EngulfSize * Constants.ENGULF_SIZE_RATIO_REQ)
+        {
+            OnNoticeMessage?.Invoke(this,
+                new SimpleHUDMessage(TranslationServer.Translate("NOTICE_ENGULF_SIZE_TOO_SMALL")));
         }
```
Comment (Kasterisk) [inline src/microbe_stage/Microbe.Contact.cs:1779]: This is also fired when touching friendly species, which shouldn't happen as that implies cannibalism.

File: src/microbe_stage/MicrobeStage.tscn [hunk 4]
```diff
@@ -2215,7 +2223,6 @@ __meta__ = {
 }
 
 [node name="PauseMenu" parent="." instance=ExtResource( 11 )]
-visible = false
```
Comment (Kasterisk) [inline src/microbe_stage/MicrobeStage.tscn:2218]: Ditto.

File: src/microbe_stage/gui/HUDMessages.tscn [hunk 5]
```diff
@@ -0,0 +1,13 @@
+[gd_scene load_steps=3 format=2]
+
+[ext_resource path="res://src/microbe_stage/gui/HUDMessages.cs" type="Script" id=1]
+[ext_resource path="res://src/gui_common/fonts/Lato-Bold-Small.tres" type="DynamicFont" id=2]
+
+[node name="HUDMessages" type="VBoxContainer"]
+pause_mode = 1
+anchor_right = 1.0
+anchor_bottom = 1.0
```
Comment (Kasterisk) [inline src/microbe_stage/gui/HUDMessages.tscn:9]: Should be anchored to the center top.

File: src/microbe_stage/gui/HUDMessages.cs [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+            SizeFlagsHorizontal = (int)SizeFlags.ExpandFill,
+            Align = Label.AlignEnum.Center,
+            Autowrap = true,
+            MouseFilter = MouseFilterEnum.Ignore,
+            Text = TextForMessage(message),
+        };
+
+        label.AddFontOverride("font", MessageFont);
+
+        AddChild(label);
```
Comment (Kasterisk) [inline src/microbe_stage/gui/HUDMessages.cs:107]: Could also have an option for messages to be stacked top-bottom here, where new messages are placed topmost first. I'd like to reuse this class for the kill feed messages in multiplayer.

Here's for example how I did it: https://github.com/Revolutionary-Games/Thrive/blob/e9d8965cdaa35f7f58f6e54ac4458ff8461ed353/src/microbe_stage/multiplayer/microbial_arena/MicrobialArenaHUD.cs#L166-L167

**Review by Kasterisk — CHANGES_REQUESTED** [review]

File: locale/en.po (hunk 1 above)
Comment (hhyyrylainen) [reply locale/en.po:2280]: Yeah, that makes sense. When quickly writing out all the English text I didn't even notice this.

File: src/microbe_stage/gui/HUDMessages.tscn (hunk 5 above)
Comment (hhyyrylainen) [reply src/microbe_stage/gui/HUDMessages.tscn:9]: Well I specifically anchored it on the whole screen as it kind of makes sense that the HUD messages can get extra space.

Comment (Kasterisk) [reply src/microbe_stage/gui/HUDMessages.tscn:9]: Okay, either one works I guess.

File: src/microbe_stage/Microbe.Contact.cs (hunk 3 above)
Comment (hhyyrylainen) [reply src/microbe_stage/Microbe.Contact.cs:1779]: I'll refactor the engulf check method to return an enum which explains why it fails so the conditions aren't repeated a bunch here.

File: src/microbe_stage/MicrobeStage.tscn (hunk 4 above)
Comment (hhyyrylainen) [reply src/microbe_stage/MicrobeStage.tscn:2218]: Actually this is the expected result. The pause menu is hidden by default, which means Godot should not write the visible property at all (as it is the default value). I recently accepted a PR that added this explicit `visible = false` here, but it seems that Godot will just remove that automatically. So I think we should just keep this change.

File: src/microbe_stage/gui/HUDMessages.cs (hunk 6 above)
Comment (hhyyrylainen) [reply src/microbe_stage/gui/HUDMessages.cs:107]: That'll probably need other changes as well, but I'll add a reverse order option to the HUDMessages class.

File: locale/en.po [hunk 7]
```diff
@@ -2275,10 +2275,14 @@ msgstr "HP:"
 msgid "HSV"
 msgstr "HSV"
 
+#: ../src/microbe_stage/gui/HUDMessages.cs:37
+msgid "HUD_MESSAGE_MULTIPLE"
+msgstr "{0} (x{1})"
```
Comment (Kasterisk) [inline locale/en.po:2280]: Should `SPECIES_N_TIMES` be replaced with this?

**Review by Kasterisk — DISMISSED** [review]

File: locale/en.po (hunk 7 above)
Comment (hhyyrylainen) [reply locale/en.po:2280]: I think the usage context is quite different, which is why I did not combine these. For example in Finnish if the "x2" wasn't understandable, different substitute words would make sense to use in these two different contexts.

**Pull request merged**